api/backend.go

```diff
@@ -564,37 +585,46 @@ func (b *StatusBackend) SelectAccount(walletAddress, chatAddress, password strin
 			return err
 		}
 
-		if err := st.InitProtocolWithPassword(chatAddress, password); err != nil {
+		if err := st.InitProtocolWithPassword(loginParams.ChatAddress.String(), loginParams.Password); err != nil {
 			return err
 		}
 	}
-	err = b.startWallet(password)
+
+	err = b.startWallet(loginParams.Password)
 	if err != nil {
 		return err
 	}
-	err = b.startBrowsers(password)
+
+	err = b.startBrowsers(loginParams.Password)
 	if err != nil {
 		return err
 	}
-	return b.startPermissions(password)
+
+	return b.startPermissions(loginParams.Password)
 }
 
 func (b *StatusBackend) startWallet(password string) error {
 	if !b.statusNode.Config().WalletConfig.Enabled {
 		return nil
 	}
+
 	wallet, err := b.statusNode.WalletService()
 	if err != nil {
 		return err
 	}
-	account, err := b.accountManager.SelectedWalletAccount()
+
+	watchAddresses := b.accountManager.WatchAddresses()
+	mainAccountAddress, err := b.accountManager.MainAccountAddress()
 	if err != nil {
 		return err
 	}
-	path := path.Join(b.statusNode.Config().DataDir, fmt.Sprintf("wallet-%x.sql", account.Address))
+
+	allAddresses := append(watchAddresses, mainAccountAddress)
```

Comment:
it will extend original b.accountManager.watchAddress if internal arrays capacity is bigger than current length.

i would suggest to copy watch address before modifying them

Reply:
good catch thank you @dshulyak